### PR TITLE
5267 tag sources

### PR DESCRIPTION
**Pull request by andresfib**
This pull requests includes also 5398 and has been rebased on develop.

### CODE REVIEW

File: public/angular/templates/sources/show.html [hunk 1]
```diff
@@ -31,6 +31,10 @@
       </td>
     </tr>
     <tr>
+      <td class="detail-key" translate>Tags</td>
+      <td class="detail-value">{{ tagsToString(source.tags) }}</td>
+    </tr>
```
Comment (pgarrison) [inline public/angular/templates/sources/show.html]: Trailing whitespace here

File: shared/tags.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Library for handling tags
+'use strict';
+var _ = require('lodash');
+
+// Tags are string comma separated in the front end
+var toArray = function(tags) {
+  if (typeof tags === 'string') {
```
Comment (pgarrison) [inline shared/tags.js]: So if it's not a string, we just return `undefined`? Won't that cause problems harder to debug than `object doesn't have method 'trim'`?

File: test/backend/models.query.report-query.test.js [hunk 3]
```diff
@@ -23,6 +51,10 @@ describe('Query attributes', function() {
     var similar = ReportQuery.compare(otherQuery, query);
     expect(similar).to.be.true;
   });
+
```
Comment (pgarrison) [inline test/backend/models.query.report-query.test.js]: Trailing whitespace here

File: public/angular/templates/sources/modal.html [hunk 4]
```diff
@@ -74,6 +74,14 @@ <h4 class="modal-title">
               </div>
             </div>
           </div>
+          <div class="form-group" ng-show="source.media">
+            <label for="tags" class="col-lg-2 control-label" translate>Tags</label>
+            <div class="col-lg-8">
+              <input ng-model="source.tags" ng-list class="form-control" id="tags" name="tags" placeholder="{{ 'Enter tags' | translate }}">
+              </input>
+              <div class="help-block" translate="sources_modal.tags"></div>
```
Comment (pgarrison) [inline public/angular/templates/sources/modal.html]: This string needs to be added to `translations/locale-en.json`

File: public/angular/templates/sources/index.html [hunk 5]
```diff
@@ -14,6 +14,7 @@
       <th class="compact" translate>Name</th>
       <th class="compact" ng-show="currentUser.can('view other users')" translate>Creator</th>
       <th class="expand" translate>Target</th>
+      <th class="expand" translate>Tags</th>
```
Comment (pgarrison) [inline public/angular/templates/sources/index.html]: This column is really skinny in my browser. Should we change that?

File: shared/tags.js (hunk 2 above)
Comment (andresfib) [inline shared/tags.js]: Good point

File: public/angular/templates/sources/modal.html (hunk 4 above)
Comment (pgarrison) [inline public/angular/templates/sources/modal.html]: Fixed

File: public/angular/templates/sources/index.html (hunk 5 above)
Comment (pgarrison) [inline public/angular/templates/sources/index.html]: Documented as issue 5446

**Pull request merged**